crates/freeze/src/datasets/logs.rs

```diff
@@ -284,5 +297,226 @@ async fn logs_to_df(
+    pub raw: String,
+    /// decoded abi type of event signature string
+    pub event: abi::Event,
+}
+
+impl LogDecoder {
+    /// create a new LogDecoder from an event signature
+    /// ex: LogDecoder::new("event Transfer(address indexed from, address indexed to, uint256
+    /// amount)".to_string())
+    pub fn new(event_signature: String) -> Option<Self> {
+        match HumanReadableParser::parse_event(event_signature.as_str()) {
+            Ok(event) => Some(Self { event, raw: event_signature.clone() }),
+            Err(_) => {
+                eprintln!("incorrectly formatted event {} (expect something like event Transfer(address indexed from, address indexed to, uint256 amount)", event_signature);
+                None
+            }
+        }
+    }
+
+    fn field_names(&self) -> Vec<String> {
+        self.event.inputs.iter().map(|i| i.name.clone()).collect()
+    }
+
+    /// converts from a log type to an abi token type
+    /// this function assumes all logs are of the same type and skips fields if they don't match the
+    /// passed event definition
+    pub fn parse_log_from_event(&self, logs: Vec<Log>) -> HashMap<String, Vec<Token>> {
+        let mut map: HashMap<String, Vec<Token>> = HashMap::new();
+        let known_keys =
+            self.event.inputs.clone().into_iter().map(|i| i.name).collect::<HashSet<String>>();
+
+        for log in logs {
+            match self.event.parse_log(RawLog::from(log)) {
+                Ok(log) => {
+                    for param in log.params {
+                        if known_keys.contains(param.name.as_str()) {
+                            let tokens = map.entry(param.name).or_insert(Vec::new());
+                            tokens.push(param.value);
+                        }
+                    }
+                }
+                Err(e) => eprintln!("error parsing log: {:?}", e),
+            }
+        }
+        map
+    }
+
+    /// data should never be mixed type, otherwise this will return inconsistent results
+    pub fn make_series(name: String, data: Vec<Token>, chunk_len: usize) -> Result<Series, String> {
+        // This is a smooth brain way of doing this, but I can't think of a better way right now
+        let mut ints: Vec<u64> = vec![];
+        let mut str_ints: Vec<String> = vec![];
+        let mut bytes: Vec<String> = vec![];
+        let mut bools: Vec<bool> = vec![];
+        let mut strings: Vec<String> = vec![];
+        let mut addresses: Vec<String> = vec![];
+        // TODO: support array & tuple types
+
+        for token in data.clone() {
+            match token {
+                Token::Address(a) => addresses.push(format!("{:?}", a)),
+                Token::FixedBytes(b) => bytes.push(b.encode_hex()),
+                Token::Bytes(b) => bytes.push(b.encode_hex()),
+                // LogParam and Token both don't specify the size of the int, so we have to guess.
```

Comment:
`EventParam` specifies the int size so could get it by passing the `abi::event` to `make_series`

Reply:
to simplify we could do something like
- for any uints <=64 bits use u64
- for any uints > 64 bits use u256

if we want even more simplicity, I think we should cast it all to u256 because u256 is so common